### DISCUSSION TITLE
How to set and get data attributes on inputMask (or inputText) components?

### DISCUSSION THREAD

**Discussion by wlannistamps** [primefaces/community#2064] [PrimeReact] [answered]
I am creating an `<InputMask>` component. I set up the ref using `useRef` and passed it in, standard stuff.
```
const ccNumberRef = useRef(null);

<InputMask
		className="credit-card-number__input form-control"
		data-qa="credit-card-number-input"
		id="ccNumber"
		mask="9999 9999 9999 9999"
		name="ccNumber"
		onBlur={(event) => inputBlur(event)}
		onChange={(event) => inputChange(event)}
		onFocus={(event) => inputFocus(event)}
		ref={ccNumberRef}
	/>
```

When I focus on the field, I want to set its current value as a data attribute, 
when I blur I want to compare the field's value to what was sent, 
and if there was a change I want to do stuff with it.
```
const inputFocus = (event) => {
	const field = event.target;

	/*
	 * Track the value on focus using dataset.onFocusVal and the value of the field.
	 * We'll use this to detect a change on blur to do stuff with the new value
	 */
	field.dataset.onFocusVal = field.value;
};

const inputBlur = (event) => {
	const field = event.target;

	/*
	 * Set the value to pass on to the store:
	 */
	const val = field.value;

	const onFocusVal = field.dataset.onFocusVal;

	/*
	 * Check if the value changed from dataset.onFocusVal using onFocusVal
	 * If so, trigger the tracking function
	 * then reset the dataset.onFocusVal to null
	 */
	if (field.dataset.onFocusVal !== field.value) {
		trackFieldComplete({
			fieldName: field.name,
			fieldValue: val
		});
	}
};
```
However, `field.dataset` does not exist. When I console the `event.target` I get an object with some of the props from the `InputMask` like the `id`, `name`, and `value`, but none of the data attributes.

When I view the input with the inspector, i can see that it's getting the attributes. How do I read them using `event`?

I saw the [Pass Through documentation](https://primereact.org/passthrough/) and it is unclear and unhelpful. There are no good examples of this. Anyone know how to actually use this?

**Comment by wlannistamps** [comment]
Ok I gave up on the PassThrough stuff. I used a few different event properties to get this to work:

- `event.nativeEvent` for Focus
- `event.originalEvent.nativeEvent` for onChange
- `event` for onBlur
This just felt like a weird hack for something I would have expected would just work, but I did see in the issues in github that slowly other components are getting their data attributes implemented. Do I need to make a request?

**Comment by wlannistamps** [comment]
If anyone knows how to get the pt object working for data attributes, that would be great though. I saw in another answer how someone had formatted theirs, maybe it would look like this?
```
<InputMask
		pt={{ input: { root: { 
			dataQa: "credit-card-number-input",
			dataPrivate: "redact"
		}}}}
		... etc />
		
```

**Reply by melloware** [reply, marked as answer]
Just tested and this works fine.

```xml
      <InputMask
        value={value}
        onChange={(e) => setValue(e.target.value)}
        mask="99-999999"
        placeholder="99-999999"
        pt={{
          root: {
            'data-qa': 'credit-card-number-input',
            'data-private': 'redact',
          },
        }}
      />
```

See it in action: https://stackblitz.com/edit/y9ha3m?file=src%2FApp.jsx
